compaction/compaction_manager.cc

```diff
@@ -925,6 +925,19 @@ void compaction_manager::submit(replica::table* t) {
     (void)perform_task(make_shared<regular_compaction_task>(*this, t));
 }
 
+bool compaction_manager::can_perform_regular_compaction(replica::table* t) {
+    if (!can_proceed(t) || t->is_auto_compaction_disabled_by_user()) {
+        return false;
+    }
```

Comment:
@bhalevy could you please explain a bit more on the motivation for this patch? do we need to mimick the locking for submitting a regular compaction job? I think that will be hard to keep in sync over time. for example, we also have the weight serialization which can prevent a job from kicking in. isn't it enough to have the timer if we decided that we should block flushes? if L0 sstables is huge, then it's not a bad idea to wait for time T before proceeding with flush, that can give compaction some time to keep up.

Reply:
My main worry is what happens during major compaction.
It might me tied up for a long time compacting the high tiers.

I think that on the long term we may want to allow regular compaction of new sstable during major compaction of the existing sstable set.  Plus, start a targeted compaction job for the inflated buckets maybe even with higher shares than refular compaction to reduce the number of sstables, ir alternatively make sure regular compaction picks those buckets next time it considers sstables for compaction (and wait for it)